kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/OperationSupport.java

```diff
@@ -173,12 +170,19 @@ public URL getResourceUrl() throws MalformedURLException {
   }
 
 
-  protected <T> void handleDelete(T resource) throws ExecutionException, InterruptedException, KubernetesClientException, IOException {
-    handleDelete(getResourceUrl(checkNamespace(resource), checkName(resource)));
+  protected <T> void handleDelete(T resource, Long gracePeriodSeconds) throws ExecutionException, InterruptedException, KubernetesClientException, IOException {
+    handleDelete(getResourceUrl(checkNamespace(resource), checkName(resource)), gracePeriodSeconds);
   }
 
-  protected void handleDelete(URL requestUrl) throws ExecutionException, InterruptedException, KubernetesClientException, IOException {
-    Request.Builder requestBuilder = new Request.Builder().delete(null).url(requestUrl);
+  protected void handleDelete(URL requestUrl, Long gracePeriodSeconds) throws ExecutionException, InterruptedException, KubernetesClientException, IOException {
+    RequestBody requestBody = null;
+    if (gracePeriodSeconds >= 0) {
```

Comment:
Is 0 a valid grace period value? Guess that's the default meaning delete immediately?